### ISSUE TITLE
Latest "What's New"s need PR links

### ISSUE THREAD

**Issue by trexfeathers** [SciTools/iris#3818]
## 📚 Documentation

Following https://github.com/SciTools/iris/pull/3816#discussion_r484087699, as many entries as possible in [docs/iris/src/whatsnew/latest.rst](https://github.com/SciTools/iris/blob/master/docs/iris/src/whatsnew/latest.rst) should have links added to the associated Iris PR(s). It might also be wise to make [the documentation note about adding such links](https://github.com/SciTools/iris/blob/master/docs/iris/src/developers_guide/documenting/whats_new_contributions.rst#L32-L35) a bit more insistent than the current "optionally you may also include..."

**Comment by bjlittle** [comment]
@trexfeathers Would it be possible to also include the following under this task:

- Add the associated GitHub author that contributed to delivering the content of the `whatsnew`, if possible
- Update the [documentation](https://scitools-iris.readthedocs.io/en/latest/developers_guide/documenting/whats_new_contributions.html#writing-a-contribution) to state that, apart from referencing the relevant issues/pull-request of a `whatsnew` entry, the GitHub author is also referenced and cited.

**Comment by trexfeathers** [comment]
> Add the associated GitHub author that contributed to delivering the content of the whatsnew

@bjlittle 👍👍👍 great idea! This was actually discussed at SciPy 2020 as an important thing to do for healthy open source communities, and I notice it's already happening in [Cartopy's "What's New"](https://scitools.org.uk/cartopy/docs/latest/whats_new.html).

**Comment by bjlittle** [comment]
@trexfeathers Cool!

Also, with regards to referencing the associated PR... as a standard to follow do you think it's best to have something like:
```
* whatsnew contribution. See :pull:`3818`.
```
Or, akin to `cartopy`:
```
* whatsnew contribution. (:pull:`38181)
```
I kinda like the latter 🤔

**Comment by trexfeathers** [comment]
I agree on the latter, which leaves us free to use `See :pull:`3818` and other syntaxes for more specific extra detail ("for extra info go check this out")

**Comment by bjlittle** [comment]
> I agree on the latter, which leaves us free to use `See :pull:` and other syntaxes for more specific extra detail ("for extra info go check this out")

@trexfeathers Awesome. Let's go with that then.

I'm currently making some `whatsnew` entries in my `cube-arithmetic-post` feature branch, so I'll adopt that style also to save any follow-up re-work.

And I'm assuming that you're going with the same approach as `cartopy` for citing authors e.g.,
```
John Smith contributed blah. (:pull:`xxxx`)
```

**Comment by trexfeathers** [comment]
Closed by #3838
